src/serde.rs

```diff
@@ -81,11 +83,62 @@ impl Write for Value {
                 }
                 Ok(())
             }
+            Self::Map(m) => {
+                if m.is_empty() {
+                    return Err(ImprintError::SchemaError("empty map not allowed".into()));
+                }
```

Comment:
let's start with the length instead of the type codes so we can support empty maps (I'll do the same for arrays) -- basically don't serialize the type codes for empty maps and jsut return early

Reply:
870bb53